src/ui/atoms/fileInput/FileInput.tsx

```diff
@@ -0,0 +1,172 @@
+    (event: LabelDragEvent): File[] =>
+      Array.from(event.dataTransfer.items)
+        .map((item: DeepReadonly<DataTransferItem>) => item.getAsFile())
+        .filter((file: DeepReadonly<File> | null): file is File => !!file),
+    []
+  )
+
+  const handleChange = useCallback(
+    // lint rule bypassed because of type 'Element' is not compatible with readonly
+    // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const filesList = event.target.files
+      const files = filesList ? [...filesList] : []
+      if (files.length) {
+        onDropped(files)
+      }
+    },
+    [onDropped]
+  )
+
+  const handleDragOver = useCallback(
+    // lint rule bypassed because of type 'Element' is not compatible with readonly
+    // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+    (event: LabelDragEvent) => {
+      preventDefaultAndStopPropagation(event)
+    },
+    [preventDefaultAndStopPropagation]
+  )
+
+  const handleDragEnter = useCallback(
+    // lint rule bypassed because of type 'Element' is not compatible with readonly
+    // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+    (event: LabelDragEvent) => {
+      preventDefaultAndStopPropagation(event)
+      addDraggingClassName()
+    },
+    [preventDefaultAndStopPropagation, addDraggingClassName]
+  )
+
+  const handleDragLeave = useCallback(
+    // lint rule bypassed because of type 'Element' is not compatible with readonly
+    // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+    (event: LabelDragEvent) => {
+      preventDefaultAndStopPropagation(event)
+      removeDraggingClassName()
+    },
+    [preventDefaultAndStopPropagation, removeDraggingClassName]
+  )
+
+  const handleDrop = useCallback(
+    // lint rule bypassed because of type 'Element' is not compatible with readonly
+    // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+    async (event: LabelDragEvent) => {
+      preventDefaultAndStopPropagation(event)
+      removeDraggingClassName()
+      onDropped(getFilesFromEvent(event))
+    },
+    [preventDefaultAndStopPropagation, removeDraggingClassName, getFilesFromEvent, onDropped]
+  )
+
+  return (
+    <div className={`okp4-fileinput-main ${size}`} ref={inputFileElement}>
+      <label
+        className={classNames('okp4-fileinput-container', { error })}
+        onDragEnter={handleDragEnter}
+        onDragLeave={handleDragLeave}
+        onDragOver={handleDragOver}
+        onDrop={handleDrop}
+      >
+        <input
+          accept={acceptedFormats?.join(', ')}
+          multiple={multiple}
+          onChange={handleChange}
+          type="file"
+        />
+        <div>
+          <img src={fileInputIcon} />
+          <Typography as="div" fontSize={size === 'small' ? 'small' : 'medium'} fontWeight="light">
+            {label}
```

Comment:
We need to be careful when rendering optional properties: if prop is `undefined`, there is still an HTML element present in the DOM which includes CSS rules such as `margin`, `padding` which can make the User eXperience a bit strange...

Reply:
Is it better to write it like that : 

```tsx
<div>
          <img src={fileInputIcon} />
          {label && (
            <Typography
              as="div"
              fontSize={size === 'small' ? 'small' : 'medium'}
              fontWeight="light"
            >
              {label}
            </Typography>
          )}
```

Reply:
@JoeyDjr 
I will say yes. Even though `JSX` is heavier, this prevents from unnecessary DOM elements